.github/workflows/style.yml

```
@@ -0,0 +1,24 @@
+# Run style checks
+
+name: style
+
+on:
+  - pull_request
+  - workflow_dispatch
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          cache: "pip"
+          cache-dependency-path: ".github/workflows/qc-requirements.txt"
+      - run: python -m pip install -r .github/workflows/qc-requirements.txt
```

We should only need `ruff` here, rather than all of the packages listed in `qc-requirements`:

```suggestion
      - run: python -m pip install ruff
```